drivers/src/stm32f446xx_gpio_driver_course.c

```diff
@@ -175,24 +183,33 @@ uint16_t GPIO_ReadInputPORT(GPIO_RegDef_t* pGPIO)
 /* GPIO Write Output PIN */
 void GPIO_WriteOutputPIN(GPIO_RegDef_t* pGPIO, uint8_t PinNumber, uint8_t value)
 {
-	if(value == GPIO_PIN_ENABLE)
+	if(value == PIN_SET_ENABLE)
 	{
-		pGPIO->ODR |= (1 << PinNumber);
+		pGPIO->BSRR |= (1 << PinNumber);
 	}
-	else if(value == GPIO_PIN_DISABLE)
+	else if(value == PIN_SET_DISABLE)
 	{
-		pGPIO->ODR &= ~(1 << PinNumber);
+		pGPIO->BSRR |= (1 << (PinNumber + 16));
 	}
 }
 
 /* GPIO Write Output Port */
 void GPIO_WriteOutputPORT(GPIO_RegDef_t* pGPIO, uint16_t value)
 {
-	pGPIO->ODR = value;
+	pGPIO->BSRR |= value; //set the requested BS bits to 1
 }
 
 /* GPIO Toogle Output PIN */
 void GPIO_ToggleOutputPIN(GPIO_RegDef_t* pGPIO, uint8_t PinNumber)
 {
-	pGPIO->ODR ^= (0x00000001 << PinNumber);
+	uint32_t temp;
+	temp = (pGPIO->ODR >> PinNumber) & 0x00000001;
```

Comment:
Why are you using the 32bits hex? You can just write 0x1, or 1. Best option is 1U or (1U << 0ul)